src/commands/force/source/open.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import * as path from 'path';
+import * as open from 'open';
+import { getBoolean, getString } from '@salesforce/ts-types';
+import { fs, AuthInfo } from '@salesforce/core';
+import { SfdxCommand, flags, FlagsConfig } from '@salesforce/command';
+import { Messages, sfdc, SfdxError } from '@salesforce/core';
+import checkLightningDomain from '@salesforce/core/lib/util/checkLightningDomain';
+import { SourceComponent, MetadataResolver } from '@salesforce/source-deploy-retrieve';
+import { OpenResultFormatter, OpenCommandResult } from '../../../formatters/openResultFormatter';
+
+export interface DnsLookupObject {
+  address: string;
+  family: number;
+}
+
+export interface FlexiPageRecord {
+  attributes: {
+    type: string;
+    url: string;
+  };
+  Id: string;
+}
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-source', 'open');
+
+export class Open extends SfdxCommand {
```

Comment:
can we make this extend `SourceCommand`?

Reply:
yes we can.